src/server/routes/signin.js

```diff
@@ -44,7 +44,7 @@ export default async function signin(req, res) {
     // according to RFC 2821, but in practice this causes more problems than
     // it solves. We treat email addresses as all lower case. If anyone
     // complains about this we can make strict RFC 2821 compliance an option.
-    const email = req.body.email?.toLowerCase() ?? null
+    const email = req.body.email.toLowerCase()
```

Comment:
There's an `await` below on **line 49** that is not part of a `try {} catch {}`. Maybe it makes sense to extend the `try {} catch {}` below to also cover these two lines:
```ts
// If is an existing user return a user object (otherwise use placeholder)
const profile = (await getUserByEmail(email)) || { email }
const account = { id: provider.id, type: "email", providerAccountId: email }
```
In this way, if there's an error here, we can do the redirect as well rather than let the error just bubble up 👍🏽

Reply:
This whole email signin function is encapsulated in a try..catch at a higher level, so I *think* it's unnecessary:

https://github.com/nextauthjs/next-auth/blob/1fb308a6f4c5d6b01cd3ba9f104264d59a270e4f/src/server/routes/signin.js#L75-L80